src/main/java/org/spongepowered/api/item/inventory/transaction/InventoryTransactionResult.java

```diff
@@ -138,6 +139,13 @@ static InventoryTransactionResult failNoTransactions() {
      */
     List<ItemStackSnapshot> getRejectedItems();
 
+    /**
+     * If items were polled by the operation this will return the polled item.
+     *
+     * @return the item that was polled from the inventory.
+     */
+    Optional<ItemStackSnapshot> getPolledItem();
```

Comment:
I prefer subclasses of `InventoryTransactionResult` instead of putting everything here.

Reply:
That would break the ability to combine multiple InventoryTransactionResults (`InventoryTransactionResult#and`)

On that note it has also to be `List<ItemStackSnapshot> getPolledItems()`

Reply:
Ok, you can still use subclasses in this case, because the moment you start merging things `getPolledItem` is meaningless. Which of the polled items is it, the first one, or the last one. So the subclass provides the `getPolledItem`, and the merged transactions (or other operations) use the standard one.

Another thing, I can see it becoming quite annoying that you have to call `getPolledItem` after calling `poll`. Most of the time users don't even care about the transactions that happened, just about the returned item.

Reply:
can't have everything. but the subclass is here